### ISSUE TITLE
Possible data leakage via in-context examples

### ISSUE THREAD

**Issue by silver-seashell** [eddie-euijun-hwang/SpaMo#12]
Hi @eddie-euijun-hwang,
Thank you so much for sharing this repository.
 
While adapting the code to my own dataset, I noticed a potential data leakage risk: in `FlanT5SLT.prepare_inputs`, the `ex_lang_trans` examples are appended to the prompt whenever `use_in_context=True`. Since this happens unconditionally, it also applies during validation and testing. Currently, the only way to avoid this is to use different configs for training and evaluation and setting `use_in_context` to False and `num_in_context` to 0 for testing.

To make this safer by default, I suggest a small patch at https://github.com/eddie-euijun-hwang/SpaMo/blob/aee7f8b6ea8201b710690adcce6f46260dd11bba/spamo/t5_slt.py#L201

```

if self.use_in_context and split == "train":

```

Best regards and thanks again for open sourcing this code base.

**Comment by XieYingchun** [comment]
> Hi [@eddie-euijun-hwang](https://github.com/eddie-euijun-hwang), Thank you so much for sharing this repository.
> 
> While adapting the code to my own dataset, I noticed a potential data leakage risk: in , the examples are appended to the prompt whenever . Since this happens unconditionally, it also applies during validation and testing. Currently, the only way to avoid this is to use different configs for training and evaluation and setting to False and to 0 for testing.`FlanT5SLT.prepare_inputs``ex_lang_trans``use_in_context=True``use_in_context``num_in_context`
> 
> To make this safer by default, I suggest a small patch at
> 
> [SpaMo/spamo/t5_slt.py](https://github.com/eddie-euijun-hwang/SpaMo/blob/aee7f8b6ea8201b710690adcce6f46260dd11bba/spamo/t5_slt.py#L201)
> 
> Line 201 in [aee7f8b](/eddie-euijun-hwang/SpaMo/commit/aee7f8b6ea8201b710690adcce6f46260dd11bba)
> 
>  if self.use_in_context: 
> ```
> 
> if self.use_in_context and split == "train":
> ```
> 
> Best regards and thanks again for open sourcing this code base.

I have been working on implementing this model and had to modify some code to get it running properly. Using the feature files and parameter files published by the original authors, I obtained the results shown in the attached image.
I was wondering if you would be willing to share your complete code implementation with me? I would greatly appreciate your help.
Thank you very much for your time and consideration.

<img width="1411" height="848" alt="Image" src="https://github.com/user-attachments/assets/75b112ab-c2a1-49c4-a24f-30d4be9c372d" />

**Comment by dksanyal** [comment]
> Hi [@eddie-euijun-hwang](https://github.com/eddie-euijun-hwang), Thank you so much for sharing this repository.
> 
> While adapting the code to my own dataset, I noticed a potential data leakage risk: in `FlanT5SLT.prepare_inputs`, the `ex_lang_trans` examples are appended to the prompt whenever `use_in_context=True`. Since this happens unconditionally, it also applies during validation and testing. Currently, the only way to avoid this is to use different configs for training and evaluation and setting `use_in_context` to False and `num_in_context` to 0 for testing.
> 
> To make this safer by default, I suggest a small patch at
> 
> [SpaMo/spamo/t5_slt.py](https://github.com/eddie-euijun-hwang/SpaMo/blob/aee7f8b6ea8201b710690adcce6f46260dd11bba/spamo/t5_slt.py#L201)
> 
> Line 201 in [aee7f8b](/eddie-euijun-hwang/SpaMo/commit/aee7f8b6ea8201b710690adcce6f46260dd11bba)
>  if self.use_in_context: 
> 
> ```
> 
> if self.use_in_context and split == "train":
> ```
> 
> Best regards and thanks again for open sourcing this code base.

A few related observations: (1) It is mentioned in the paper "To ensure that the LLM does not directly access the target translations during training, we shuffle the translation samples so that they do not match the target translation." However, in the code, it seems that for a given input video, the target sentence along with its translations in one or more of the following languages, English, French and Spanish, are included. Thus, there seems to be a disconnect between the code and the paper. (2) Moreover, the purpose of these in-context examples is also unclear to me since the examples teach textual translation while the model is trying to learn sign-to-spoken language translation. (3) The authors also mention in the paper "At test time, we select a translation pair from the training set to use as a reference." It would be great if the authors share the code for the test configuration.
